common/rfb/VNCServerST.cxx

```diff
@@ -313,6 +313,8 @@ void VNCServerST::setPixelBuffer(PixelBuffer* pb_, const ScreenSet& layout)
   screenLayout = layout;
 
   if (!pb) {
+    stopFrameClock();
```

Comment:
Perhaps it is better to do this when the desktop stops? I.e. in `VNCServerST::removeSocket()`.

Reply:
I have placed it there because it is symetric with the `startFrameClock()` - it is called when pixel buffer is set, so `stopFrameClock()` is called when pixel buffer is unset.

Currently the call chain is `VNCServerST::removeSocket()` -> `rfb::SDesktop::stop()` (in this case x0vncserver's `XDesktop::stop`) -> `VNCServerST::setPixelBuffer(0)`. But other implementation of `rfb::SDesktop::stop` may not want to unset the pixel buffer there, so the timer would stop and not start again for next connection? (Not sure.)

Reply:
Right, other server may not call `VNCServerST::setPixelBuffer(0)`. However that won't wedge things as the clock will start again once new changes happen. The only downside is that the clock will continue to run for a bit after the desktop is "stopped", which might be a bit confusing. My idea was to make it more symmetrical with `SDesktop::start()/stop()`.